packages/edit-site/src/components/sidebar-dataviews/dataview-item.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * WordPress dependencies
+ */
+import { page, columns } from '@wordpress/icons';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import { useLink } from '../routes/link';
+import SidebarNavigationItem from '../sidebar-navigation-item';
+import { unlock } from '../../lock-unlock';
+const { useLocation } = unlock( routerPrivateApis );
+
+function getDataViewIcon( type ) {
+	const icons = { list: page, grid: columns };
+	return icons[ type ];
+}
+
+export default function DataViewItem( {
+	title,
+	slug,
+	customViewId,
+	type,
+	icon,
+	isActive,
+	isCustom,
+} ) {
+	const {
+		params: { path },
+	} = useLocation();
+
+	const iconToUse = icon || getDataViewIcon( type );
+
+	const linkInfo = useLink( {
+		path,
+		activeView: isCustom === 'true' ? customViewId : slug,
```

Comment:
It's super confusing to have `isCustom` as a string. How did this come to be? Is it because of how params are expected by `useLink`? Or is it due to how values are parsed from template records? Either way, shouldn't we be casting to boolean and vice-versa? If not, this prop should probably have a less ambiguous name. /cc @jorgefilipecosta